src/components/utils/ComboBoxItems/ComboBoxItems.tsx

```diff
@@ -52,39 +58,54 @@ function ComboBoxItems<Item>({
 
   return (
     <>
-      {Object.entries(groupedItems).map(([groupName, items]) => (
-        <Box aria-label={`Group ${groupName}`} as="li" key={groupName} listStyle="none">
-          <Text
-            position="sticky"
-            top={0}
-            pl={4}
-            py={4}
-            zIndex={1}
-            fontWeight="bold"
-            textTransform="uppercase"
-            backgroundColor="navyblue-350"
-          >
-            {groupName}
-          </Text>
-          <Box as="ul" role="listbox">
-            {items.map(item => (
+      {Object.entries(groupedItems).map(([groupName, items]) => {
+        const enabledItems = items.filter(i => !disableItem(i));
+        const disabled = enabledItems.length === 0;
+
+        return (
+          <Box aria-label={`Group ${groupName}`} as="li" key={groupName} listStyle="none">
+            <Box position="sticky" top={0} textTransform="uppercase" fontWeight="bold" zIndex={1}>
               <MenuItem
-                {...rest.getItemProps({ item, disabled: rest.disableItem(item) })}
-                as="li"
-                listStyle="none"
-                nested
-                key={rest.itemToString(item)}
+                {...(allowMultipleSelection && {
+                  ...getItemProps({
```

Comment:
Using `getItemProps` from `Downshift` here.
This way the group item is treated as a MenuItem from Downshift. With this change users can:
- Navigate and select group items using keyboard arrows.
- Highlight group items without additional hacks(Downshift does that for us internally)
- Select the group (the value that will be returned will be an array containing all the `enabled` items in the group)